plugins/etcd/src/main/java/org/radargun/service/EtcdClientOperations.java

```diff
@@ -0,0 +1,135 @@
+      this.service = service;
+   }
+
+   @Override
+   public Cache<byte[], byte[]> getCache(String cacheName) {
+      return new EtcdCacheAdapter(service.kvClient);
+   }
+
+   private class EtcdCacheAdapter implements BasicOperations.Cache<byte[], byte[]> {
+
+      private KV kvClient;
+
+      public EtcdCacheAdapter(KV kvClient) {
+         this.kvClient = kvClient;
+      }
+
+      @Override
+      public byte[] get(byte[] key) {
+         Optional<KeyValue> value = this.retrieveValueFrom(key);
+         return retrieveBytes(value);
+      }
+
+      @Override
+      public boolean containsKey(byte[] key) {
+         Optional<KeyValue> value = this.retrieveValueFrom(key);
+         return value.isPresent();
+      }
+
+      @Override
+      public void put(byte[] key, byte[] value) {
+         try {
+            kvClient.put(ByteSequence.fromBytes(key), ByteSequence.fromBytes(value)).get();
+         } catch (InterruptedException | ExecutionException e) {
+            throw new IllegalStateException("Impossible to put the cache value", e);
+         }
+      }
+
+      @Override
+      public byte[] getAndPut(byte[] key, byte[] value) {
+         PutResponse response;
+         try {
+            response = kvClient.put(ByteSequence.fromBytes(key), ByteSequence.fromBytes(value), GET_PUT_OPTION).get();
+         } catch (InterruptedException | ExecutionException e) {
+            throw new IllegalStateException("Impossible to put the cache value", e);
+         }
+
+         if (response.hasPrevKv()) {
+            return response.getPrevKv().getValue().getBytes();
+         } else {
+            return null;
+         }
+      }
+
+      @Override
+      public boolean remove(byte[] key) {
+         DeleteResponse response;
+         try {
+            response = kvClient.delete(ByteSequence.fromBytes(key)).get();
+         } catch (InterruptedException | ExecutionException e) {
+            throw new IllegalStateException("Impossible to delete the cache", e);
+         }
+         return response.getDeleted() > 0;
+      }
+
+      @Override
+      public byte[] getAndRemove(byte[] key) {
+         DeleteResponse response;
+         try {
+            response = kvClient.delete(ByteSequence.fromBytes(key), GET_DELETE_OPTION).get();
+         } catch (InterruptedException | ExecutionException e) {
+            throw new IllegalStateException("Impossible to delete the cache", e);
+         }
+
+         List<KeyValue> prevKvs = response.getPrevKvs();
+         if (prevKvs != null && prevKvs.size() > 0) {
+            return prevKvs.get(0).getValue().getBytes();
+         } else {
+            return null;
+         }
+      }
+
+      @Override
+      public void clear() {
+         throw new UnsupportedOperationException("Clearing etcd is not supported.");
+      }
+
+      private byte[] retrieveBytes(Optional<KeyValue> optionalKeyValue) {
+         if (!optionalKeyValue.isPresent()) {
```

Comment:
When you use `Optional` you should rather use `.orElseThrow(...).getValue().getBytes()`.